### ISSUE TITLE
Unexpected: Git diffing doesn't take into account unstaged files

### ISSUE THREAD

**Issue by jamesbraza** [promptfoo/promptfoo-action#122]
I am manually changing my `promptfooconfig.yaml` in CI. This means the change is not committed, it's a local change.

`promptfoo-action` v1's `git diff` here: https://github.com/promptfoo/promptfoo-action/blob/v1/src/main.ts#L101-L105

It doesn't take into account local changes as leading to changed files. Thus my `promptfoo-action` doesn't invoke `promptfoo` and instead emits:

```none
No LLM prompt or config files were modified.
```

Can we make this diff account for local file changes as well?

**Comment by typpo** [comment]
Interesting. Wouldn't it be pretty unusual for files to be changed locally
on a stateless ci runner?

**Comment by jamesbraza** [comment]
Yeah I think in general you're right that it is unusual, so feel free to close this out.

To share, my use case is generating the `promptfooconfig.yaml` on the fly in CI: https://github.com/Future-House/LitQA/blob/ba2b3f9f9798b8c7f40632c9ec3d68b3c1a6c67f/.github/create_tests.py

**Comment by mldangelo** [comment]
Thanks @jamesbraza. Use `force-run: true` for generated configs:

```yaml
- uses: promptfoo/promptfoo-action@v1
  with:
    config: 'promptfooconfig.yaml'
    force-run: true
```

Closing as `force-run` handles this.
